src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -79,35 +79,33 @@ public BitArray(byte[] bytes)
                 throw new ArgumentException(SR.Format(SR.Argument_ArrayTooLarge, BitsPerByte), nameof(bytes));
             }
 
-            m_array = new int[GetArrayLength(bytes.Length, BytesPerInt32)];
+            m_array = new int[GetInt32ArrayLengthFromByteLength(bytes.Length)];
             m_length = bytes.Length * BitsPerByte;
 
-            int i = 0;
-            int j = 0;
-            while (bytes.Length - j >= 4)
+            uint totalCount = (uint)bytes.Length / 4;
+
+            ReadOnlySpan<byte> byteSpan = bytes;
```

Comment:
Suggestion: try the below single line instead of the loop.

```cs
MemoryMarshal.Cast<byte, int>(bytes).CopyTo(m_array);
```

Reply:
What about the left over bytes that don't fit into an int? I tried the following but it fails several tests:
```C#
int totalCount = Div4Rem(bytes.Length, out int leftOver);

ReadOnlySpan<int> intSpan = MemoryMarshal.Cast<byte, int>(bytes);
intSpan.CopyTo(m_array);

int last = 0;
switch (leftOver)
{
    case 3:
        last = bytes[bytes.Length - 3] << 16;
        goto case 2;
    // fall through
    case 2:
        last |= bytes[bytes.Length - 2] << 8;
        goto case 1;
    // fall through
    case 1:
        m_array[totalCount] = last | bytes[bytes.Length - 1];
        break;
}
```

Reply:
That improves the performance for large lengths, but regresses small length ctor.